tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py

```diff
@@ -345,6 +347,287 @@ def create_tf_saved_model_dir(temp_dir):
+            res['result'] = tf.sigmoid(tf.nn.relu(args['args']))
+            return res
+    return LayerModel()
+
+def create_keras_layer_with_example_input_1(tmp_dir):
+    import tensorflow as tf
+    model = create_keras_layer_input_list()
+
+    example_input = (np.random.rand(1,2,3).astype(np.float32), np.random.rand(1,2,3).astype(np.float32))
+
+    input_shapes = [PartialShape([1,2,3]),
+                    PartialShape([1,2,3])]
+
+    param1 = ov.opset8.parameter(input_shapes[0], dtype=np.float32)
+    param2 = ov.opset8.parameter(input_shapes[1], dtype=np.float32)
+    add = ov.opset8.add(param1, param2)
+    relu1 = ov.opset8.relu(add)
+    sigm1 = ov.opset8.sigmoid(relu1)
+    relu2 = ov.opset8.relu(param1)
+    sigm2 = ov.opset8.sigmoid(param2)
+
+    parameter_list = [param1, param2]
+    model_ref = Model([sigm1, relu2, sigm2], parameter_list, "test")
+
+    return model, model_ref, {'example_input': example_input}
+
+
+def create_keras_layer_with_example_input_2(tmp_dir):
+    import tensorflow as tf
+    model = create_keras_layer_input_dict()
+
+    example_input = {'a': np.random.rand(1,2,3).astype(np.float32), 'b': np.random.rand(1,2,3).astype(np.float32)}
+
+
+    input_shapes = [PartialShape([1,2,3]),
+                    PartialShape([1,2,3])]
+
+    param1 = ov.opset8.parameter(input_shapes[0], dtype=np.float32)
+    param2 = ov.opset8.parameter(input_shapes[1], dtype=np.float32)
+    add = ov.opset8.add(param1, param2)
+    relu1 = ov.opset8.relu(add)
+    sigm1 = ov.opset8.sigmoid(relu1)
+
+    parameter_list = [param1, param2]
+    model_ref = Model([sigm1], parameter_list, "test")
+
+    return model, model_ref, {'example_input': example_input}
+
+
+def create_keras_layer_with_input_shapes_case1(tmp_dir):
+    import tensorflow as tf
+    model = create_keras_layer_input_list()
+
+    input_shapes = [PartialShape([1,2,3]),
+                    PartialShape([1,2,3])]
+
+    param1 = ov.opset8.parameter(input_shapes[0], dtype=np.float32)
+    param2 = ov.opset8.parameter(input_shapes[1], dtype=np.float32)
+    add = ov.opset8.add(param1, param2)
+    relu1 = ov.opset8.relu(add)
+    sigm1 = ov.opset8.sigmoid(relu1)
+    relu2 = ov.opset8.relu(param1)
+    sigm2 = ov.opset8.sigmoid(param2)
+
+    parameter_list = [param1, param2]
+    model_ref = Model([sigm1, relu2, sigm2], parameter_list, "test")
+
+    return model, model_ref, {'input_shape': [[1, 2, 3], [1, 2, 3]]}
+
+
+def create_keras_layer_with_input_shapes_case2(tmp_dir):
+    import tensorflow as tf
+    model = create_keras_layer_input_list()
+
+    input_shapes = [PartialShape([1,2,3]),
+                    PartialShape([1,2,3])]
+
+    param1 = ov.opset8.parameter(input_shapes[0], dtype=np.float32)
+    param2 = ov.opset8.parameter(input_shapes[1], dtype=np.float32)
+    add = ov.opset8.add(param1, param2)
+    relu1 = ov.opset8.relu(add)
+    sigm1 = ov.opset8.sigmoid(relu1)
+    relu2 = ov.opset8.relu(param1)
+    sigm2 = ov.opset8.sigmoid(param2)
```

Comment:
I see a lot of duplication here. Can we have for each test model also function for generation of ov::Model reference. Or one function that returns the original model and reference model

Reply:
Done.